temporaltest/server_test.go

```diff
@@ -0,0 +1,407 @@
+		RegisterWorkflowsAndActivities(registry)
+	})
+
+	// Start test workflow
+	wfr, err := c.ExecuteWorkflow(
+		context.Background(),
+		client.StartWorkflowOptions{TaskQueue: "hello_world"},
+		Greet,
+		"world",
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Get workflow result
+	var result string
+	if err := wfr.Get(context.Background(), &result); err != nil {
+		t.Fatal(err)
+	}
+
+	// Print result
+	fmt.Println(result)
+	// Output: Hello world
+}
+
+func TestNewServer(t *testing.T) {
+	ts := temporaltest.NewServer(temporaltest.WithT(t))
+
+	ts.NewWorker("hello_world", func(registry worker.Registry) {
+		RegisterWorkflowsAndActivities(registry)
+	})
+
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	wfr, err := ts.GetDefaultClient().ExecuteWorkflow(
+		ctx,
+		client.StartWorkflowOptions{TaskQueue: "hello_world"},
+		Greet,
+		"world",
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var result string
+	if err := wfr.Get(ctx, &result); err != nil {
+		t.Fatal(err)
+	}
+
+	if result != "Hello world" {
+		t.Fatalf("unexpected result: %q", result)
+	}
+}
+
+func TestNewWorkerWithOptions(t *testing.T) {
+	ts := temporaltest.NewServer(temporaltest.WithT(t))
+
+	ts.NewWorkerWithOptions(
+		"hello_world",
+		func(registry worker.Registry) {
+			RegisterWorkflowsAndActivities(registry)
+		},
+		worker.Options{
+			MaxConcurrentActivityExecutionSize:      1,
+			MaxConcurrentLocalActivityExecutionSize: 1,
+		},
+	)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	wfr, err := ts.GetDefaultClient().ExecuteWorkflow(
+		ctx,
+		client.StartWorkflowOptions{TaskQueue: "hello_world"},
+		Greet,
+		"world",
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var result string
+	if err := wfr.Get(ctx, &result); err != nil {
+		t.Fatal(err)
+	}
+
+	if result != "Hello world" {
+		t.Fatalf("unexpected result: %q", result)
+	}
+
+}
+
+func TestDefaultWorkerOptions(t *testing.T) {
+	ts := temporaltest.NewServer(
+		temporaltest.WithT(t),
+		temporaltest.WithBaseWorkerOptions(
+			worker.Options{
+				MaxConcurrentActivityExecutionSize:      1,
+				MaxConcurrentLocalActivityExecutionSize: 1,
+			},
+		),
+	)
+
+	ts.NewWorker("hello_world", func(registry worker.Registry) {
+		RegisterWorkflowsAndActivities(registry)
+	})
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	wfr, err := ts.GetDefaultClient().ExecuteWorkflow(
+		ctx,
+		client.StartWorkflowOptions{TaskQueue: "hello_world"},
+		Greet,
+		"world",
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var result string
+	if err := wfr.Get(ctx, &result); err != nil {
+		t.Fatal(err)
+	}
+
+	if result != "Hello world" {
+		t.Fatalf("unexpected result: %q", result)
+	}
+}
+
+type denyAllClaimMapper struct{}
+
+func (denyAllClaimMapper) GetClaims(*authorization.AuthInfo) (*authorization.Claims, error) {
+	return nil, fmt.Errorf("no claims for you!")
```

Comment:
an error is a bit different from "no claims", but I guess in this case has the same effect

Reply:
Yep I suppose I could rename this to failAllClaimMapper. The goal is only to verify that additional server options are applied.